nntrainer/compiler/tflite_opnode.cpp

```diff
@@ -25,13 +25,41 @@ TfOpNode::TfOpNode() :
   is_input(false),
   is_output(false),
   node_owned_variable(),
+  /// @todo distinguish between uninitialized and ADD operator.
   op_type(tflite::BuiltinOperator_ADD),
   builtin_ops(),
   builtin_option_type(tflite::BuiltinOptions_NONE){};
 
 void TfOpNode::setLayerNode(const LayerNode &layer) {
   is_input = layer.getNumInputConnections() == 0;
   is_output = layer.getNumOutputConnections() == 0;
+  /// @todo support more loss layers
+  static const std::set<std::string> loss_type = {"mse", "cross"};
+  /** set to graph output node if output connection of the node includes loss
+   *layer string
+   *  @note this is workaround because it cannot be guaranteed that a loss layer
+   *always has a loss type in its name.
+   *
+   *  There are two ways to pass `GraphRepresentation` parameters to `serialize`
+   *method.
+   *
+   *  1. with loss layer at the end of the graph
+   *  2. wihtout loss layer but last node has loss layer output connection
```

Comment:
FYI, I don't know about the steps before the `serialize`. So, I had to print them manually.

```c++
void TfliteInterpreter::serialize(const GraphRepresentation &representation,
                                  const std::string &out) {
  /// @todo check if graph is finalized & initialized and ready to serialize.

  for (auto x : representation){
    std::cout << x->getName() << std::endl;
  }
```

```bash
# run resnet
./nntrainer_resnet18 fake 1 128 1
started computation at Tue May 24 11:18:29 2022

data_dir: fake batch_size: 1 data_split: 128 epoch: 1
#1/1 - Training Loss: 0.00985765 >> [ Accuracy: 0% - Validation Loss : 0.0117535 ]
input0
conv0
first_bn_relu
first_bn_relu/activation_realized
first_bn_relu/generated_out_0
conv1_0/a1
conv1_0/a2
conv1_0/a2/activation_realized
conv1_0/a3
conv1_0/c1
conv1_0
conv1_0/activation_realized
conv1_0/generated_out_0
conv1_1/a1
conv1_1/a2
conv1_1/a2/activation_realized
conv1_1/a3
conv1_1/c1
conv1_1
conv1_1/activation_realized
conv1_1/generated_out_0
conv2_0/b1
conv2_0/a1
conv2_0/a2
conv2_0/a2/activation_realized
conv2_0/a3
conv2_0/c1
conv2_0
conv2_0/activation_realized
conv2_0/generated_out_0
conv2_1/a1
conv2_1/a2
conv2_1/a2/activation_realized
conv2_1/a3
conv2_1/c1
conv2_1
conv2_1/activation_realized
conv2_1/generated_out_0
conv3_0/b1
conv3_0/a1
conv3_0/a2
conv3_0/a2/activation_realized
conv3_0/a3
conv3_0/c1
conv3_0
conv3_0/activation_realized
conv3_0/generated_out_0
conv3_1/a1
conv3_1/a2
conv3_1/a2/activation_realized
conv3_1/a3
conv3_1/c1
conv3_1
conv3_1/activation_realized
conv3_1/generated_out_0
conv4_0/b1
conv4_0/a1
conv4_0/a2
conv4_0/a2/activation_realized
conv4_0/a3
conv4_0/c1
conv4_0
conv4_0/activation_realized
conv4_0/generated_out_0
conv4_1/a1
conv4_1/a2
conv4_1/a2/activation_realized
conv4_1/a3
conv4_1/c1
conv4_1
conv4_1/activation_realized
last_p1
last_f1
fully_connected0
fully_connected0/activation_realized # When I printed its output connection, loss layer is printed 

```
As you can see, there is no loss layer but last node has output connection with loss layer. Am I wrong?

Reply:
This is because loss layer is added at compile phase not relization when given as a model property.

https://github.com/nnstreamer/nntrainer/blob/895d73d822977b9078c9f8e56fe243336ed8ece9/nntrainer/models/neuralnet.cpp#L152-L163

Extracting graph representation in from model_graph will do the job although I guess the propose code also do the job